File: sources/lua/15_file_write/main.lua
-- task 15 file_write — expected output: 104857600
-- build: none (interpreted)    run: lua main.lua (PUC Lua 5.4)    also runs on LuaJIT: luajit main.lua
-- Buffer is the 1 MiB pattern 0,1,2,...,255 repeated 4096 times, written 100 times.
-- Lua's standard library has no fsync; the file is flushed and closed instead.

local part = {}
for i = 0, 255 do
    part[i + 1] = string.char(i)
end
local buf = string.rep(table.concat(part), 4096)

local f = io.open("out.bin", "wb")

local written = 0
for _ = 1, 100 do
    f:write(buf)
    written = written + #buf
end

f:flush()
f:close()
print(written)

File: sources/julia/15_file_write/main.jl
# task 15 file_write — expected output: 104857600
# build: julia main.jl    run: julia main.jl
# deviation: Julia's standard library exposes no fsync, so the file is flushed and closed.

function main()
    buf = Vector{UInt8}(undef, 1048576)
    for i in 1:1048576
        buf[i] = UInt8((i - 1) % 256)
    end

    written = Int64(0)
    open("out.bin", "w") do f
        for _ in 1:100
            written += write(f, buf)
        end
        flush(f)
    end
    println(written)
end

main()
